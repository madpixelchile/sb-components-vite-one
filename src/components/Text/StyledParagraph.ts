
import styled from "styled-components";
import breakpoints from "../../styles/breakpoints";

export const StyledParagraph = styled.p`

    ${({ theme }) => {
        const { fontConfig, colors } = theme;

        return `

            font-family: ${fontConfig["font-family-base"]};
            line-height: ${fontConfig["line-height-base"]};
            font-size: ${fontConfig["font-size-text-6"]};

            @media only screen and (min-width: ${breakpoints["screen-lg"]}) {
                font-size: ${fontConfig["font-size-base"]};
            }

            span{
                color: ${colors['color-span']};
            }

        `;
    }}

`;