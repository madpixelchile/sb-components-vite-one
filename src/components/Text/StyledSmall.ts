
import styled from "styled-components";
import fontConfig from "../../styles/fontConfig";

export const StyledSmall = styled.small`
    font-size: ${fontConfig["font-size-text-8"]};
`;

