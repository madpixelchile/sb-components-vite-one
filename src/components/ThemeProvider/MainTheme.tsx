

import { ThemeProvider } from 'styled-components';
import { StyledThemeProvider } from './StyledThemeProvider';
import { darkMode, lightMode } from '../../styles/theme';

export interface MainThemeProps {
  children?: React.ReactNode;
  theme?: string | {},
  [key: string]: any;
}

export const MainTheme = ({ children = null, theme = 'light' }:MainThemeProps) => {

  let selectedTheme: string | object = lightMode;

  if(typeof theme !== 'object'){
    if(theme === 'light'){
      //No object - string light - default lightModeAdded
      selectedTheme = lightMode;
    }else if(theme === 'dark'){
      //No object - string dark -  default darkModeAdded
      selectedTheme = darkMode;
    }
  }else if(typeof theme === 'object'){
    if(Object.keys(theme).length === 0){ //!PENDING Compare with the final BASE required properties
      //Empty object - Default lightMode added
      selectedTheme = lightMode;
    }else{
      //Full object - CustomMode or theme added
      selectedTheme = theme;
    }
  }

  return (
    <>
        <ThemeProvider theme={ selectedTheme }>
            <StyledThemeProvider/>
            { children }
        </ThemeProvider>
    </>
  )
}

export default  MainTheme;
