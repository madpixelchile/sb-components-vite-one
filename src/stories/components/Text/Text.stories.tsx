import { Meta, StoryObj } from "@storybook/react";
import {Text, TextProps} from "../../../components/Text/Text";

const meta = {
    //Se recomienda generar un storybook a parte del proyecto final y con vite
    //El slash genera la visualización en carpetas y subniveles en la navegación principal
    title: 'UI/text/Text',
    //El componente a incorporar en el muestrario
    component: Text,

    tags:[
        'autodocs'
    ],
    //Para centrar el componente en la mesa o cama (muestrario) de componente en pantalla
    parameters: {
        layout: 'centered'
    },
    //Para que en la documentación los tamaños salgan en formato radio y hacia el lado
    argTypes:{
        variant: { control: 'select', options: ['p','span','strong','small', 'em'] },
        children: { control: 'text' }
    },

} satisfies Meta<typeof Text>;

export default meta;

//Ejemplo de componente con sus props específicas aplicadas
export const BasicText = ({ children, ...props }:TextProps)=>(
    <Text {...props} >{ children }</Text>
)


