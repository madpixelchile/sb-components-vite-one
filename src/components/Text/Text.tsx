import React from "react";
import { StyledParagraph } from "./StyledParagraph";
import { StyledSpan } from "./StyledSpan";
import { StyledSmall } from "./StyledSmall";

export interface TextProps {
    children?: React.ReactNode;
    variant?: string;
    label?: string;
    [key: string]: any;
}

export const Text: React.FC<TextProps> = (
    {
        children = null,
        variant = 'p',
        label = 'Example text',
        ...props
    }
) => {
    return (
        <>
            { variant === 'p' && <StyledParagraph {...props}>{children || label}</StyledParagraph> }
            { variant === 'span' && <StyledSpan {...props}>{children || label}</StyledSpan> }
            { variant === 'small' && <StyledSmall {...props}>{children || label}</StyledSmall> }
            { variant === 'strong' && <strong {...props}>{children || label}</strong> }
            { variant === 'em' && <em {...props}>{children || label}</em> }
        </>
    )
}

export default Text;

