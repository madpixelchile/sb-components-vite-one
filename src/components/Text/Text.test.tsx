
import { render } from '@testing-library/react'; //screen screen.debug();
import { Text } from './Text';
import MainTheme from '../ThemeProvider/MainTheme';

describe('Testing Texts', () => {

    test('Testing prop children and className', () => {

        const { getByText, container } = render(<MainTheme><Text className='h2'>Children</Text></MainTheme>);
        const childrenElement = getByText('Children');
        expect(childrenElement.innerHTML).toBe('Children');

        const classProp = container.getElementsByClassName('h2');
        expect(classProp.length).toBe(1)

        expect(container).toMatchSnapshot();

    })

    test('Testing prop variant span', () => {
        const { container } = render(<MainTheme><Text variant={'span'}></Text></MainTheme>);
        expect(container.querySelector('div span')?.innerHTML).toBe('Texto de ejemplo');
    })

    test('Testing prop variant small', () => {
        const { container } = render(<MainTheme><Text variant={'small'}></Text></MainTheme>);
        expect(container.querySelector('div small')?.innerHTML).toBe('Texto de ejemplo');
    })

    test('Testing prop label without children', () => {
        const { container } = render(<MainTheme><Text variant={'em'} label={'Hola mundo'} ></Text></MainTheme>);
        expect(container.querySelector('div em')?.innerHTML).toBe('Hola mundo');
    })

    test('Testing prop without content', () => {
        const { container } = render(<MainTheme><Text variant={'strong'} ></Text></MainTheme>);
        expect(container.querySelector('div strong')?.innerHTML).toBe('Texto de ejemplo');
    })

    test('Testing prop variant p', () => {
        const { container } = render(<MainTheme><Text></Text></MainTheme>);
        expect(container.querySelector('div p')?.innerHTML).toBe('Texto de ejemplo');
    })

})