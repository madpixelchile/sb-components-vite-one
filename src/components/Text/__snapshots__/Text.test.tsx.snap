// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Texts Testing prop children and className 1`] = `
<div>
  <p
    class="sc-bdnyFh gQCwAX h2"
  >
    Children
  </p>
</div>
`;
